woocommerce/api/CacheableRequestTrait.php

```diff
@@ -0,0 +1,93 @@
+<?php
+/**
+ * WooCommerce Plugin Framework
+ *
+ * This source file is subject to the GNU General Public License v3.0
+ * that is bundled with this package in the file license.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://www.gnu.org/licenses/gpl-3.0.html
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@skyverge.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade the plugin to newer
+ * versions in the future. If you wish to customize the plugin for your
+ * needs please refer to http://www.skyverge.com
+ *
+ * @package   SkyVerge/WooCommerce/API/Response
+ * @author    SkyVerge
+ * @copyright Copyright (c) 2013-2021, SkyVerge, Inc.
+ * @license   http://www.gnu.org/licenses/gpl-3.0.html GNU General Public License v3.0
+ */
+
+namespace SkyVerge\WooCommerce\PluginFramework\v5_10_10\API;
+
+trait CacheableRequestTrait {
+
+	/** @var int the cache lifetime for the request, in seconds, defaults to 0 (unlimited) */
+	protected $cache_lifetime = 0;
```

Comment:
I just realized that defining the property in the trait makes it impossible for classes using this property to use property assignment. If a class wants to override the default `$cache_lifetime`, it would need to do so in the constructor, or on a per-instance basis by using `$request->set_cache_lifetime()`.

I don't really fancy the idea of having to defining the default cache lifetime of a request class in it's constructor. However, I'm not excited about not enforcing the property to be protected by the trait, either. Afterall, if the trait does not define the property as protected, then the class using this trait would have free reign over whether it defines the property as protected, public, private or whether it defines the property at all.

As such, I'm torn.

Reply:
Also, I'm not sure if an unlimited cache lifetime by default is reasonable. This might fill up the `wp_options` table rather quickly on busy sites.

Reply:
I went back and forth with these 2 things. In the end, I think it makes sense to provide a sane default that won't blow up anybody's database. With regards to the `$cache_lifetime` property - I think having to assign the property in the constructor is the lesser evil considering the alternative, which is to leave it undefined and up to the class using this trait to define it. The issue with that is that there's nothing enforcing the property to be defined.

Reply:
>I just realized that defining the property in the trait makes it impossible for classes using this property to use property assignment.

Yeah I often forget about that! I concur that the best experience we can have is to set it in the constructor, and that's what we've been doing in the other platform code as well. I wish we could redefine the value outright at the top of the class, but the constructor is an alright alternative and better than having things like `property_exists` checks, etc...